### PR TITLE
Add TypeUtteranceEndResponse & handle case in live.router

### PR DESCRIPTION
**Pull request by boyswan**
## Proposed changes

Reference: https://github.com/deepgram/deepgram-go-sdk/issues/146

## Types of changes

What types of changes does your code introduce to the community Go SDK?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update or tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/api/live/v1/router.go [hunk 1]
```diff
@@ -53,6 +53,8 @@ func (r *MessageRouter) Message(byMsg []byte) error {
 		return r.MessageResponse(byMsg)
 	case interfaces.TypeMetadataResponse:
 		return r.MetadataResponse(byMsg)
+	case interfaces.TypeUtteranceEndResponse:
+		return r.UtteranceEndResponse(byMsg)
```
Comment (dvonthenen) [inline pkg/api/live/v1/router.go:57]: Is appears that there is some byte stream that is returned from the Deepgram platform here. Can this be turned into a Go Struct?

**Review by dvonthenen — CHANGES_REQUESTED** [review]

File: pkg/api/live/v1/interfaces/interfaces.go [hunk 2]
```diff
@@ -9,6 +9,7 @@ package interfaces
 type LiveMessageCallback interface {
 	Message(mr *MessageResponse) error
 	Metadata(md *MetadataResponse) error
+	UtteranceEnd() error
```
[the remark this answers is not in the document]
Comment (boyswan) [reply pkg/api/live/v1/interfaces/interfaces.go:12]: So this is something I wanted to discuss. There is actually no data in the byMsg, as the UtteranceEnd message contains nothing but `{ type: "UtteranceEnd" }`. Arguably I shouldn't be passing in byMsg at all - however I was wondering if you would prefer to have a `UtteranceEndReponse` empty struct for consistency?

[the remark this answers is not in the document]
Comment (dvonthenen) [reply pkg/api/live/v1/interfaces/interfaces.go:12]: Yea, I think for compatibility sake, we probably want a structure even if it's empty.

The idea is that additional information could be passed back at some point and we can forward that onto the user. That change would be non-breaking and therefore it's better to have an empty structure than nothing at all. Users currently implementing where no information is passed back. Isn't looking at values in the structure anyways.

**Comment by dvonthenen** [comment]
Thanks for the update. I'll take a look at this in a little bit! I super appreciate it!

**Review by dvonthenen — APPROVED** [review]

**Pull request merged**